backend/dataall/modules/datasets/services/dataset_service.py

```diff
@@ -50,7 +51,67 @@
 log = logging.getLogger(__name__)
 
 
+class DatasetServiceInterface(ABC):
+    @staticmethod
+    def check_before_delete(session, uri, **kwargs) -> bool:
+        """Abstract method to be implemented by dependent modules that want to add checks before deletion for dataset objects"""
+        return True
+
+    @staticmethod
+    def execute_on_delete(session, uri, **kwargs) -> bool:
+        """Abstract method to be implemented by dependent modules that want to add clean-up actions when a dataset object is deleted"""
+        return True
+
+    @staticmethod
+    def append_to_list_user_datasets(session, username, groups):
+        """Abstract method to be implemented by dependent modules that want to add datasets to the list_datasets that list all datasets that the user has access to"""
+        return True
+
+    @staticmethod
+    def resolve_additional_dataset_user_role(session, uri, username, groups):
+        """Abstract method to be implemented by dependent modules that want to add new types of user role in relation to a Dataset"""
+        return None
+
+    @staticmethod
+    def extend_attach_steward_permissions(session, dataset, new_stewards) -> bool:
+        """Abstract method to be implemented by dependent modules that want to attach additional permissions to Dataset stewards"""
+        return True
+
+    def extend_delete_steward_permissions(session, dataset, new_stewards) -> bool:
+        """Abstract method to be implemented by dependent modules that want to attach additional permissions to Dataset stewards"""
+        return True
```

Comment:
Should we use [@abstractmethod](https://docs.python.org/3/library/abc.html#abc.abstractmethod) and `...` for unfinished code?

Reply:
My concern is that not all implementations of DatasetServiceInterface might use all methods. But for that the solution would be to break down the interface into smaller interfaces following (Interface Segregation principle)